samples/renderer/src/winmain.cpp

```diff
@@ -0,0 +1,211 @@
+            wolf_object.addComponent<rcbe::rendering::Material>(std::move(material));
+            wolf_object.addComponent<rcbe::visual::VisualTextureSet>(std::move(vts));
+        }
+
+        {
+            std::unordered_set<rcbe::visual::TexturePtr> tex_cache;
+            rcbe::rendering::Material::MaterialConfig mc {
+                {{"datamodel/data/rendering/shaders/viking_room.vert.spv",
+                  rdmn::render::ShaderType::vertex,
+                  rdmn::render::ShaderState::precompiled},
+                 {"datamodel/data/rendering/shaders/viking_room.frag.spv",
+                  rdmn::render::ShaderType::fragment,
+                  rdmn::render::ShaderState::precompiled}}};
+
+            rcbe::rendering::Material m(std::move(mc));
+            auto mat_ptr = std::make_shared<rcbe::core::CoreObject>(std::move(m));
+
+            for (size_t i = 0; i < viking_room_objects.size(); ++i) {
+                auto &c = viking_room_objects[i];
+                const auto &tex_component = c.getComponent<rdmn::render::RasterizerTexture>();
+                if (!tex_component)
+                    continue;
+                auto &rt = tex_component->as<rdmn::render::RasterizerTexture>();
+                const auto &vtex_ptr = rt.getVisualTexturePtr();
+
+                const auto it = tex_cache.find(vtex_ptr);
+                if (it == tex_cache.end()) {
+                    const auto [vtex_it, res] = tex_cache.insert(vtex_ptr);
+
+                    auto &mat = mat_ptr->as<rcbe::rendering::Material>();
+                    auto inserted_tex_it = mat.pushTexture(
+                        std::make_shared<rdmn::render::RasterizerTexture>(
+                            rt.getConfig(),
+                            rt.getVisualTexturePtr()));
+
+                    if (!c.hasComponent<rcbe::rendering::Material>())
+                        c.addComponent<rcbe::rendering::Material>(mat_ptr);
+                } else {
+                    if (!c.hasComponent<rcbe::rendering::Material>())
+                        c.addComponent<rcbe::rendering::Material>(mat_ptr);
+                }
+            }
+        }
+
+#ifdef RDMN_VULKAN
+        auto camera_conf = rcbe::utils::read_from_file<rcbe::rendering::camera_config>(
+            "datamodel/data/rendering/vulkan_camera_config.json");
+#elif defined(RDMN_DIRECTX)
+        auto camera_conf = rcbe::utils::read_from_file<rcbe::rendering::camera_config>(
+            "datamodel/data/rendering/default_camera_config.json");
+#else
+        static_assert(false, RASTERIZER_NOT_SET_ERROR_MSG);
+#endif
+
+        auto camera = rcbe::rendering::make_camera(window->getRenderingContext(), camera_conf);
+        auto start = std::chrono::steady_clock::now();
+        auto aim = std::make_shared<rcbe::core::AbstractInputManager>(rcbe::core::EditorInputManager::create(
+            window->getRenderingContext(), camera));
```

Comment:
I'd say it should be other way around, `InputManager` should go inside window, and be invoked from inside window.

Reply:
I'll leave a `TODO`

Reply:
Still no `TODO`